### ISSUE TITLE
Running on Windows Dev Kit (arm)

### ISSUE THREAD

**Issue by hbjoroy** [rancher-sandbox/rancher-desktop#4630]
When trying to install Rancher Desktop on Windows on Arm (using winget to install), the Rancher Desktop itself runs fine, but it looks like it tries to install amd64 k3s binaries in WSL.

I find it hard to understand the messages displayed when the problem occurs.

I guess the Rancher Desktop will be fine in emulated mode (amd64), but the point of running k3s on this machine would be to run amd64 edition in WSL2.

from the logs 
Wsl.log
---------------
2023-05-09T06:39:04.497Z: Registered distributions: Ubuntu
2023-05-09T06:39:05.183Z: Registered distributions: Ubuntu
2023-05-09T06:39:08.071Z: Registered distributions: Ubuntu,rancher-desktop
2023-05-09T06:39:08.290Z: Registered distributions: Ubuntu,rancher-desktop
2023-05-09T06:39:08.292Z: Creating initial data distribution...
2023-05-09T06:39:08.676Z: WSL: executing: /bin/busybox mkdir -p /etc/rancher: Error: wsl.exe exited with code 1
2023-05-09T06:39:08.677Z: Error registering data distribution: Error: wsl.exe exited with code 1
2023-05-09T06:39:08.685Z: WSL: executing: /bin/busybox mkdir -p /var/lib: Error: wsl.exe exited with code 1
2023-05-09T06:39:08.833Z: WSL failed to execute wsl.exe --unregister rancher-desktop-data: Error: wsl.exe exited with code 4294967295
2023-05-09T06:39:08.833Z: Error setting up data distribution: Error: wsl.exe exited with code 4294967295
    at ChildProcess.<anonymous> (C:\Users\haral\AppData\Local\Programs\Rancher Desktop\resources\app.asar\dist\app\background.js:1:8136)
    at ChildProcess.emit (node:events:527:28)
    at ChildProcess._handle.onexit (node:internal/child_process:291:12) {
  code: 4294967295,
  command: [ 'wsl.exe', '--unregister', 'rancher-desktop-data' ]
}
2023-05-09T06:39:09.034Z: WSL: executing: /bin/sh -c [ -e /etc/os-release ] && . /etc/os-release ; echo ${VERSION_ID:-0.1}: Error: wsl.exe exited with code 1

End of  k8s.log:
-----------------
2023-05-09T06:38:27.397Z: Version 0.1.0 is less than the minimum 1.15.0, skipping.
2023-05-09T06:38:27.397Z: Skipping pre-release v0.1.0-rc8
2023-05-09T06:38:27.397Z: Skipping pre-release v0.1.0-rc7
2023-05-09T06:38:27.397Z: Skipping pre-release v0.1.0-rc6
2023-05-09T06:38:27.397Z: Skipping pre-release v0.1.0-rc5
2023-05-09T06:38:27.397Z: Skipping pre-release v0.1.0-rc4
2023-05-09T06:38:27.397Z: Skipping pre-release v0.1.0-rc3
2023-05-09T06:38:27.397Z: Skipping pre-release v0.1.0-rc1
2023-05-09T06:38:27.399Z: Got 137 versions.
2023-05-09T06:39:05.129Z: Ensuring images available for K3s 1.26.4
2023-05-09T06:39:05.136Z: Will attempt to download exe https://github.com/k3s-io/k3s/releases/download/v1.26.4+k3s1/k3s to C:\Users\haral\AppData\Local\rancher-desktop\cache\k3s\tmp-v1.26.4+k3s1-qglkGN\k3s
2023-05-09T06:39:05.140Z: Will attempt to download images https://github.com/k3s-io/k3s/releases/download/v1.26.4+k3s1/k3s-airgap-images-amd64.tar.zst to C:\Users\haral\AppData\Local\rancher-desktop\cache\k3s\tmp-v1.26.4+k3s1-qglkGN\k3s-airgap-images-amd64.tar.zst
2023-05-09T06:39:05.142Z: Will attempt to download checksum https://github.com/k3s-io/k3s/releases/download/v1.26.4+k3s1/sha256sum-amd64.txt to C:\Users\haral\AppData\Local\rancher-desktop\cache\k3s\tmp-v1.26.4+k3s1-qglkGN\sha256sum-amd64.txt

**Comment by gaktive** [comment]
Hi @hbjoroy! This looks like what we want to do in https://github.com/rancher-sandbox/rancher-desktop/issues/1462, which has been around a while but it's still something we'd like to do. It's just a matter of juggling with the size of the team we have.

**Comment by jandubois** [comment]
> This looks like what we want to do in #1462,

I don't think it is; that issue is about running arm64 binaries on amd64, but this issue is about running an arm64 VM on arm64 Windows. Right now the whole VM is running in emulation mode, as we don't ship an arm64 VM. So I'm somewhat surprised that is is running at acceptable speed.

So this issue is more like building the M1 builds for macOS. I would hope that we could do cross-compilation for most pieces, but we would still need hardware for testing (something like https://www.microsoft.com/en-us/d/windows-dev-kit-2023/94k0p67w7581?activetab=pivot:overviewtab)

**Comment by SchrodingerZhu** [comment]
Any update on this? Docker desktop on windows already supports native arm64.

**Comment by jandubois** [comment]
> Any update on this?

Sorry, there is no update; it still isn't on the roadmap. 😞

**Comment by hohwille** [comment]
I have the same need. Please be aware the ARM is the new big thing in the laptop market as the Snapdragon X CPU family is so much more energy efficient and like MacOS where the revolution has already taken place and x86 is dead, on Windows and Linux it is just about happening. I bought this ARM laptop because I am an OSS developer and want to build and test my releases of software products I am maintaining also for the ARM target architecture. Currently windows on ARM feels a little bleeding edge and some software does not even run with emulation so it is not yet ideal for the mainstream user but this is only because software vendors like you and me do not properly support it yet and need to do their homework.
I hope you see a little motivation to prepare for the future and also ship ARM releaes of rancher desktop :)
Thanks for considering it...

**Comment by hohwille** [comment]
I could not even install Rancher-Desktop x86 for Windows on my Win11@ARM. The installer gets stuck forever at 0%

**Comment by ninlil** [comment]
+1
Also interested in Rancher Desktop for Windows with ARM-CPUs.

**Comment by yuriyslobodchikovrc** [comment]
+1
Also interested in Rancher Desktop for Windows with ARM-CPUs.

**Comment by jandubois** [comment]
Hint: we sometimes sort issues (for a specific area) by the number of "GitHub reactions" they got. So to get better visibility for your "+1" you should add e.g. a 👍 on the description of the issue (I don't think reactions on comments are included).

**Comment by yuriyslobodchikovrc** [comment]
> > This looks like what we want to do in [#1462](https://github.com/rancher-sandbox/rancher-desktop/issues/1462),
> 
> I don't think it is; that issue is about running arm64 binaries on amd64, but this issue is about running an arm64 VM on arm64 Windows. Right now the whole VM is running in emulation mode, as we don't ship an arm64 VM. So I'm somewhat surprised that is is running at acceptable speed.
> 
> So this issue is more like building the M1 builds for macOS. I would hope that we could do cross-compilation for most pieces, but we would still need hardware for testing (something like https://www.microsoft.com/en-us/d/windows-dev-kit-2023/94k0p67w7581?activetab=pivot:overviewtab)

Hello, link now cannot be opened

**Comment by jandubois** [comment]
> > So this issue is more like building the M1 builds for macOS. I would hope that we could do cross-compilation for most pieces, but we would still need hardware for testing (something like https://www.microsoft.com/en-us/d/windows-dev-kit-2023/94k0p67w7581?activetab=pivot:overviewtab)
> 
> [@jandubois](https://github.com/jandubois) Hello, link now cannot be opened

I think this was just a link to the Windows Dev Kit for ARM64, something like https://learn.microsoft.com/en-us/windows/arm/dev-kit/

Nowadays there are plenty of hardware choices, so this specific device is no longer relevant.

**Comment by hbjoroy** [comment]
I would also think you could install arm version of windows on arm mac parallels or other virtualisation. And you l believe you could provision arm hardware in azure cloud.

**Comment by ninlil** [comment]
This issue is now passed 2 years old, and still nothing tagged beyond being a "bug" for "windows"... really annoying.
I really love this solution when I had my previous Surface Pro, but now I have to rely on other tools that don't work as good as this used to do.

**Comment by mmattersdorfer** [comment]
+1
Also interested in Rancher Desktop for Windows with ARM-CPUs.

**Comment by rfay** [comment]
FYI GIthub does have Windows ARM runners these days, https://docs.github.com/en/actions/reference/runners/github-hosted-runners#standard-github-hosted-runners-for-public-repositories

**Comment by jandubois** [comment]
@rfay Thanks for letting me know! I didn't notice this has happened, and it removes one major stumbling block.
